controllers/user.go

```diff
@@ -577,3 +583,83 @@ func (c *ApiController) RemoveUserFromGroup() {
 	c.Data["json"] = wrapActionResponse(object.RemoveUserFromGroup(owner, name, util.GetId(owner, groupName)))
 	c.ServeJSON()
 }
+
+// SendInvite
+// @Title SendInvite
+// @router /send-invite [post]
+// @Tag User API
+func (c *ApiController) SendInvite() {
+	owner := c.Input().Get("owner")
+	username := c.Input().Get("name")
+
+	if !c.IsAdmin() {
+		c.ResponseError(c.T("auth:Unauthorized operation"))
+		return
+	}
+
+	user, err := object.GetUser(util.GetId(owner, username))
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	if !util.IsEmailValid(user.Email) {
+		c.ResponseError(fmt.Sprintf(c.T("service:Invalid Email receivers: %s"), user.Email))
+		return
+	}
+
+	organization, err := object.GetOrganization(util.GetId("admin", owner))
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	application, err := object.GetApplication(util.GetId("admin", user.SignupApplication))
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	provider, err := application.GetEmailProvider()
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+
+	if provider == nil {
+		c.ResponseError(c.T("service:Please set an Email provider first"))
+		return
+	}
+
+	sender := organization.DisplayName
+	origin := conf.GetConfigString("origin")
+
+	var link string
+	switch user.Type {
+	case "invited-user":
+		link = fmt.Sprintf("%s/signup/%s?id=%s&u=%s&e=%s", origin, application.Name, user.Id, user.Name, user.Email)
```

Comment:
does user.Email has mask?

Reply:
no, masked email require change logic at send verification code method. Is it critical?